x-pack/plugins/cross_cluster_replication/public/app/components/follower_index_form.js

```diff
@@ -333,62 +308,49 @@ export const FollowerIndexForm = injectI18n(
       /**
        * Leader index
        */
-      const renderLeaderIndex = () => {
-        const hasError = !!fieldsErrors.leaderIndex;
-        const isInvalid = hasError && areErrorsVisible;
+      const renderLeaderIndex = () => (
+        <FormEntryRow
+          field="leaderIndex"
+          value={followerIndex.leaderIndex}
+          error={fieldsErrors.leaderIndex}
+          schema={follwerIndexFormSchema.leaderIndex}
+          disabled={!isNew}
+          areErrorsVisible={areErrorsVisible}
+          onValueUpdate={this.onFieldsChange}
+          onErrorUpdate={this.onFieldsErrorChange}
+        />
+      );
 
-        return (
-          <EuiDescribedFormGroup
-            title={(
-              <EuiTitle size="s">
-                <h4>
-                  <FormattedMessage
-                    id="xpack.crossClusterReplication.followerIndexForm.sectionLeaderIndexTitle"
-                    defaultMessage="Leader index"
-                  />
-                </h4>
-              </EuiTitle>
-            )}
-            description={(
-              <Fragment>
-                <p>
-                  <FormattedMessage
-                    id="xpack.crossClusterReplication.followerIndexForm.sectionLeaderIndexDescription1"
-                    defaultMessage="The leader index you want to replicate from the remote cluster."
-                  />
-                </p>
-              </Fragment>
-            )}
-            fullWidth
+      /**
+       * Advanced settings
+       */
+      const renderAdvancedSettings = () => (
+        <Fragment>
+          <EuiButtonEmpty
+            iconType={areAdvancedSettingsVisible ? "minusInCircle" : "plusInCircle"}
+            flush="left"
+            onClick={this.toggleAdvancedSettings}
           >
-            <EuiFormRow
-              label={(
-                <FormattedMessage
-                  id="xpack.crossClusterReplication.followerIndexForm.fieldLeaderIndexLabel"
-                  defaultMessage="Leader index"
-                />
-              )}
-              helpText={(
-                <FormattedMessage
-                  id="xpack.crossClusterReplication.followerIndexForm.indexNameHelpLabel"
-                  defaultMessage="Spaces and the characters {characterList} are not allowed."
-                  values={{ characterList: <strong>{indexNameIllegalCharacters}</strong> }}
-                />
-              )}
-              isInvalid={isInvalid}
-              error={fieldsErrors.leaderIndex && fieldsErrors.leaderIndex.message}
-              fullWidth
-            >
-              <EuiFieldText
-                isInvalid={isInvalid}
-                value={leaderIndex}
-                onChange={e => this.onFieldsChange({ leaderIndex: e.target.value })}
-                fullWidth
+            { toggleAdvancedSettingButtonLabel }
+          </EuiButtonEmpty>
+          <EuiSpacer size="s" />
```

Comment:
minor: I think this could be size `m` or `l`, due to amount of whitespace above:

![image](https://user-images.githubusercontent.com/1965714/51006775-9aeac480-14f9-11e9-9316-095fb8c2e16a.png)

Reply:
Mmm I guess it's a question of taste 😊  @snide an opinion on this?

Reply:
I think it would make sense to use a similar pattern to what we have in ILM, where the section has its own heading, description, and a button for activating and deactivating the form it contains. In this case, I think it makes sense to rename the button "Use advanced settings" and "Don't use advanced settings", to make it clear that when the settings are hidden they won't be applied.

![image](https://user-images.githubusercontent.com/1238659/51156544-f639f100-1830-11e9-961f-11d27d84a082.png)

Reply:
++ to this pattern!